crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -547,6 +560,108 @@ fn finalize_nominator_deposit<T: Config>(
     Ok(())
 }
 
+pub(crate) fn do_finalize_slashed_operators<T: Config>(
+    domain_id: DomainId,
+) -> Result<(), TransitionError> {
+    let mut total_slashed_amount = BalanceOf::<T>::zero();
+    for (operator_id, slash_info) in PendingSlashes::<T>::take(domain_id).unwrap_or_default() {
+        let slashed_amount = slash_operator::<T>(operator_id, slash_info)?;
+        total_slashed_amount = total_slashed_amount
+            .checked_add(&slashed_amount)
+            .ok_or(TransitionError::BalanceOverflow)?;
+    }
+
+    // TODO: mint the total slashed amount to treasury
+
+    Ok(())
+}
+
+fn thaw_and_burn_frozen_balance<T: Config>(
+    id: &FungibleFreezeId<T>,
+    from: &T::AccountId,
+) -> Result<(), TransitionError> {
+    // thaw the frozen amount and burn
+    let locked_amount = T::Currency::balance_frozen(id, from);
+    T::Currency::thaw(id, from).map_err(|_| TransitionError::RemoveLock)?;
+    T::Currency::burn_from(from, locked_amount, Precision::Exact, Fortitude::Force)
```

Comment:
The alternative would be doing a transfer instead of burn and mint since all the slashed amount goes to the treasury, which seemingly looks easier as the reader knows what happens to the slashed stake in the first place IMHO, non-blocker.

Reply:
True. I have a new [issue](https://github.com/subspace/subspace/issues/1711) if handled, will make this change possible. I'll handle that in the next PR

Reply:
I also prefer transfer instead of burn and mint but mostly out of security concern. Because with transfer we have an automatic accounting check that the fund is right, and with burn and mint if something goes wrong (like some unconsidered rounding) it will happen silently and maybe too late when we notice.